torch/_dynamo/utils.py

```diff
@@ -3355,6 +3355,29 @@ def wrapper(*args):
     return wrapper
 
 
+@dataclasses.dataclass
+class CompiledAutogradMetrics:
+    id: int
+    start_time_ns: int
+    end_time_ns: int
+
+
+def get_compiled_autograd_metrics(maybe_gm):
```

Comment:
nit: can you add typehints to these new helpers; I think we've been getting a bit more principled about using them in this particular file at least